files/ui/graphicspage.ui

```diff
@@ -146,6 +146,70 @@
      </layout>
     </widget>
    </item>
+   <item>
+    <widget class="QGroupBox" name="gridGroupBox">
+     <property name="title">
+      <string>Distant Terrain</string>
+     </property>
+     <layout class="QGridLayout" name="gridLayout_4" columnstretch="1,0">
+      <item row="5" column="0">
+       <widget class="QCheckBox" name="distantTerrainCheckBox">
+        <property name="text">
+         <string>Enable</string>
+        </property>
+       </widget>
+      </item>
+      <item row="7" column="0">
+       <widget class="QLabel" name="smallFeatureCullingLabel">
+        <property name="text">
+         <string>Small Feature Culling</string>
+        </property>
+        <property name="alignment">
+         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
+        </property>
+       </widget>
+      </item>
+      <item row="7" column="1">
+       <layout class="QGridLayout" name="smallFeatureGridLayout">
+        <item row="0" column="1">
+         <layout class="QHBoxLayout" name="smallFeatureBoxLayout"/>
+        </item>
+        <item row="0" column="0">
+         <widget class="QCheckBox" name="smallFeatureCullingCheckBox">
+          <property name="text">
+           <string>Enable</string>
+          </property>
+         </widget>
+        </item>
+        <item row="0" column="1">
+         <widget class="QSpinBox" name="smallFeatureCullingPixelSizeSpinBox">
+          <property name="minimum">
+           <number>1</number>
+          </property>
+          <property name="maximum">
+           <number>16</number>
+          </property>
+         </widget>
+        </item>
+       </layout>
+      </item>
+      <item row="6" column="0">
+       <widget class="QLabel" name="viewingDistanceLabel">
+        <property name="text">
+         <string>Viewing Distance</string>
+        </property>
+       </widget>
+      </item>
+      <item row="6" column="1">
+       <widget class="QDoubleSpinBox" name="viewingDistanceSpinBox">
+        <property name="maximum">
+         <double>4294967296.000000000000000</double>
```

Comment:
How was this number chosen? Also, I know that viewing distance takes a float, but I'm wondering if we should make it an integer in the spinbox and just convert it to a double under the hood. Would users really want to type in a decimal for this?

Also, would it be better to offer several presets? For instance: `14285 (2 Cells)`. I'd ideally like for this to be something that people can use without having to consult the documentation.

Reply:
I suppose it is just a 2^32. I wonder why this value was chosen since spinbox value is the double anyway.

Reply:
[The documentation](https://openmw.readthedocs.io/en/master/reference/modding/settings/camera.html#viewing-distance) suggests up to 35924 (5 cells), so 2^32 seems like incredible overkill (it's over 500,000 cells).

Reply:
In theory, 10 cells should be enough. Actually, it is possible to just use an integer value in cells and then convert it to game units.

Reply:
I'd be in favor of changing this spinner to an integer number of cells, so long as it could handle the user putting a different value manually in their openmw.cfg file (perhaps as an "Other" value). For comparison, check out the screenshot extension code, which will respect the user manually putting in a different screenshot type in openmw.cfg aside from the expected ones.

Reply:
What would be the maximum cell value for the SpinBox?

Reply:
Right now I set whatever is put into the spin box will be multiplied by 8192, is this okay?